hummingbot/connector/exchange/bitrue/bitrue_api_order_book_data_source.py

```diff
@@ -0,0 +1,187 @@
+import asyncio
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from hummingbot.connector.exchange.binance.binance_order_book import BinanceOrderBook
+from hummingbot.connector.exchange.bitrue import bitrue_constants as CONSTANTS, bitrue_web_utils as web_utils
+from hummingbot.core.data_type.order_book_message import OrderBookMessage, OrderBookMessageType
+from hummingbot.core.data_type.order_book_tracker_data_source import OrderBookTrackerDataSource
+from hummingbot.core.utils.tracking_nonce import NonceCreator
+from hummingbot.core.web_assistant.connections.data_types import RESTMethod, WSJSONRequest
+from hummingbot.core.web_assistant.web_assistants_factory import WebAssistantsFactory
+from hummingbot.core.web_assistant.ws_assistant import WSAssistant
+from hummingbot.logger import HummingbotLogger
+
+if TYPE_CHECKING:
+    from hummingbot.connector.exchange.bitrue.bitrue_exchange import BitrueExchange
+
+
+class BitrueAPIOrderBookDataSource(OrderBookTrackerDataSource):
+    HEARTBEAT_TIME_INTERVAL = 30.0
+    TRADE_STREAM_ID = 1
+    DIFF_STREAM_ID = 2
+    ONE_HOUR = 60 * 60
+
+    _logger: Optional[HummingbotLogger] = None
+
+    def __init__(
+        self,
+        trading_pairs: List[str],
+        connector: "BitrueExchange",
+        api_factory: WebAssistantsFactory,
+        domain: str = CONSTANTS.DEFAULT_DOMAIN,
+    ):
+        super().__init__(trading_pairs=trading_pairs)
+        self._connector = connector
+        self._domain = domain
+        self._api_factory = api_factory
+        self._message_id_generator = NonceCreator.for_microseconds()
+        self._last_connection_check_message_sent = -1
+        self._diff_messages_queue_key = CONSTANTS.ORDERBOOK_CHANNEL_SUFFIX
+
+    async def get_last_traded_prices(self, trading_pairs: List[str], domain: Optional[str] = None) -> Dict[str, float]:
+        return await self._connector.get_last_traded_prices(trading_pairs=trading_pairs)
+
+    async def get_ticker(self, trading_pair: str) -> OrderBookMessage:
+        ticker_data = await self._get_ticker_data(trading_pair=trading_pair)
+        ticker_timestamp: float = self._time()
+        update_id = int(ticker_timestamp * 1e3)
+        ticker_msg: OrderBookMessage = BinanceOrderBook.ticker_message_from_rest_endpoint(
```

Comment:
why we use the binance order book?

Reply:
The data returned is similar to Binance's. Would you like me to duplicate the code for Bitrue?

Reply:
yes because if binance changes or Bitrue changes we are going to have an error

Reply:
Alright. Update in a minute.